test/lib/formValidator.test.ts

```diff
@@ -0,0 +1,217 @@
+import { FormValidator } from "../../src/lib/formValidator";
+
+describe("FormValidator", () => {
+  describe("validate", () => {
+    it("when given a form data object with a valid field, returns a 'valid' response", () => {
+      const formData = { testFieldId: "valid value" };
+      const mockFieldValidator = {
+        validate: jest.fn(() => {
+          return {
+            value: formData.testFieldId,
+            valid: true,
+            invalid: false,
+            errors: [],
+          };
+        }),
+      };
+      const fieldValidatorDictionary = {
+        testFieldId: mockFieldValidator,
+      };
+
+      const validationResponse = FormValidator.validate(
+        formData,
+        fieldValidatorDictionary
+      );
+
+      expect(validationResponse).toEqual({
+        testFieldId: {
+          value: "valid value",
+          valid: true,
+          invalid: false,
+          errors: [],
+        },
+      });
+    });
+
+    it("when given a form data object with an invalid field, returns an 'invalid' response", () => {
+      const formData = { anotherTestFieldId: "invalid value" };
+      const mockFieldValidator = {
+        validate: jest.fn(() => {
+          return {
+            value: formData.anotherTestFieldId,
+            valid: false,
+            invalid: true,
+            errors: ["TooLong"],
+          };
+        }),
+      };
+      const fieldValidatorDictionary = {
+        anotherTestFieldId: mockFieldValidator,
+      };
+
+      const validationResponse = FormValidator.validate(
+        formData,
+        fieldValidatorDictionary
+      );
+
+      expect(validationResponse).toEqual({
+        anotherTestFieldId: {
+          value: "invalid value",
+          valid: false,
+          invalid: true,
+          errors: ["TooLong"],
+        },
+      });
+    });
+
+    it("when given a form data object with one valid and one invalid field, returns an 'invalid' response", () => {
```

Comment:
Just a note on the tests, is that you can use `it` or `test` in Jest, there is no difference just personal preference (I personally prefer `it`).  That being said, I would recommend tests with `it` be written in a way that's readable. i.e: 

`it('should return an 'invalid' response when....')`

Should be read as `it should return an 'invalid' response when....`.  If you read the above `it when given a form data...` doesn't read correct.

Reply:
D'oh!  Rectified 👍🏻